components/AddServerForm.vue

```diff
@@ -1,55 +1,61 @@
 <template>
   <div>
-    <v-form
-      ref="form"
-      v-model="validInputs"
-      :disabled="loading"
-      @submit.prevent="connectToServer"
-    >
-      <v-text-field
-        v-model="serverUrl"
-        outlined
-        :label="$t('serverAddress')"
-        type="url"
-        :rules="rules.serverUrlTest"
-        required
-      ></v-text-field>
-      <v-row align="center" no-gutters>
-        <v-col class="mr-2">
-          <v-btn
-            :disabled="!validInputs"
-            :loading="loading"
-            block
-            large
-            color="primary"
-            type="submit"
-            >{{ $t('connect') }}</v-btn
-          >
-        </v-col>
-        <v-col cols="auto">
-          <locale-switcher />
-        </v-col>
-      </v-row>
-    </v-form>
+    <validation-observer v-slot="{ invalid }">
+      <v-form ref="form" :disabled="loading" @submit.prevent="connectToServer">
+        <validation-provider
+          v-slot="{ errors }"
+          name="serverUrl"
+          :rules="rules.serverUrl"
+        >
+          <v-text-field
+            v-model="serverUrl"
+            outlined
+            :label="$t('serverAddress')"
+            type="url"
+            :error-messages="errors"
+            required
+          ></v-text-field>
+        </validation-provider>
+        <v-row align="center" no-gutters>
+          <v-col class="mr-2">
+            <v-btn
+              :disabled="invalid"
+              :loading="loading"
+              block
+              large
+              color="primary"
+              type="submit"
+              >{{ $t('connect') }}</v-btn
+            >
+          </v-col>
+          <v-col cols="auto">
+            <locale-switcher />
+          </v-col>
+        </v-row>
+      </v-form>
+    </validation-observer>
   </div>
 </template>
 
 <script lang="ts">
 import Vue from 'vue';
 import { mapActions } from 'vuex';
+import { ValidationObserver, ValidationProvider } from 'vee-validate';
 
 export default Vue.extend({
+  components: {
```

Comment:
Generally, we're heading more towards making components globally available instead of importing them locally

You can take a look at the files in `~/plugins/components/*.ts` to see how to do that.

Reply:
My bad, I'll fix

Reply:
Do you prefer for the component to be in the components folder or can I smash everything in the plugin/veeValidate.ts file?